### PR TITLE
CB-14422 Resilient Pollers need to be implemented for Upscale operations

### PR DESCRIPTION
**Pull request by bbihari**
Add ablility to PollingService to return a result instead of throwing exception in case of timeout.

### CODE REVIEW

**Comment by horadla23** [comment]
I would add the aws e2e label to this PR to test a real cluster provision

File: core/src/main/java/com/sequenceiq/cloudbreak/core/cluster/ClusterManagerUpscaleService.java [hunk 1]
```diff
@@ -67,15 +68,21 @@ public void upscaleClusterManager(Long stackId, String hostGroupName, Integer sc
...
-            connector.waitForHosts(reachableInstances);
+            result = connector.waitForHosts(reachableInstances);
         } else {
-            connector.waitForHosts(stackService.getByIdWithListsInTransaction(stackId).getRunningInstanceMetaDataSet());
+            result = connector.waitForHosts(stackService.getByIdWithListsInTransaction(stackId).getRunningInstanceMetaDataSet());
+        }
+        if (result != null && result.isTimeout()) {
+            LOGGER.info("Upscaling cluster manager were not successful for nodes: {}", result.getFailedInstanceIds());
+            //instanceMetaDataService.updateInstanceStatus(result.getFailedInstanceIds(), InstanceStatus.ZOMBIE,
+            //        "Upscaling cluster manager were not successful.";
```
[the remark this answers is not in the document]
Comment (bbihari) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/cluster/ClusterManagerUpscaleService.java:85]: yes, there's no zombie state yet

[the remark this answers is not in the document]
Comment (horadla23) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/cluster/ClusterManagerUpscaleService.java:85]: it makes sense, is it worth to wait for Laci's PR which has the zombie status?

[the remark this answers is not in the document]
Comment (bbihari) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/cluster/ClusterManagerUpscaleService.java:85]: I think no, we will need a 3rd pr anyway

**Review by horadla23 — APPROVED** [review]

**Review by lacikaaa — APPROVED** [review]

**Pull request merged**